UnityProject/Assets/Scripts/Objects/Research/Artifacts/Effects/AreaEffects/AreaEffectBase.cs

```diff
@@ -0,0 +1,99 @@
+using System.Linq;
+using UnityEngine;
+using System.Collections.Generic;
+using HealthV2;
+
+namespace Systems.Research
+{
+	public class AreaEffectBase : ArtifactEffect
+	{
+		public EffectShapeType effectShapeType = EffectShapeType.Square;
+
+		public int coolDown = 10;
+
+		[Tooltip("How far this artifact will effect players from")]
+		public int AuraRadius = 10;
+
+		[Tooltip("Allows the artifact to effect the head and body differently if they are wearing only partially protective gear")]
+		public bool AllowBodySplitting = false;
+
+		public virtual void DoEffectAura(GameObject centeredAround)
+		{
+			var objCenter = centeredAround.AssumedWorldPosServer().RoundToInt();
+			var shape = EffectShape.CreateEffectShape(effectShapeType, objCenter, AuraRadius);
+
+			foreach (var pos in shape)
+			{
+				// check if tile has any alive player
+				var players = MatrixManager.GetAt<PlayerScript>(pos, true).Distinct();
+				foreach (var player in players)
+				{
+					if (!player.IsDeadOrGhost)
+					{
+						//What is this? Particuarly powerful artifacts can rip players apart if they are not careful,
+						//if an artifact tries to teleport the head, but the body is resistant to teleporting,
+						//it might rip the head off the body. Be careful when enabling body splitting.
+						//This can also be used to make artifacts indivdually effect body parts.
+						if (AllowBodySplitting) TryEffectParts(player);
+						else TryEffectPlayer(player);
+
+					}
+				}
+			}
+		}
+
+		protected virtual void TryEffectParts(PlayerScript toEffect)
+		{
+			foreach (BodyPart part in toEffect.playerHealth.SurfaceBodyParts)
+			{
+				if (part.BodyPartType == BodyPartType.Chest) continue; //Don't want to dismember torso, only periphirals.
+				float Armour = 0;
```

Comment:
Find better names for these armour variables. First one violates the naming convention for local variables, second one is called exactly the same and that might confuse future readers of the code.